source/server/listener_manager_impl.cc

```diff
@@ -101,13 +123,22 @@ ProdListenerComponentFactory::createListenSocket(Network::Address::InstanceConst
     return std::make_shared<Network::UdsListenSocket>(address);
   }
 
-  const std::string addr = fmt::format("tcp://{}", address->asString());
+  const std::string scheme = (socket_type == Network::Address::SocketType::Stream) ? "tcp" : "udp";
```

Comment:
optional not: think it's worth defining tcp and udp somewhere and reusing them?

Reply:
As luck would have it, turns out Network::Utility already defines these as constants, so I used those.